src/test/java/org/example/integration/CountHealthInsuranceContributionServiceTest.java

```diff
@@ -0,0 +1,258 @@
+package org.example.integration;
+
+import org.example.DataCreator;
+import org.example.adapter.out.InMemoryInvoiceRepo;
+import org.example.application.*;
+import org.example.domain.customer.Customer;
+import org.example.domain.customer.Entrepreneurship;
+import org.example.domain.customer.EntrepreneurshipForm;
+import org.example.domain.customer.TaxPayments.FlatTax;
+import org.example.domain.customer.TaxPayments.GeneralTax;
+import org.example.domain.customer.TaxPayments.IndustryType;
+import org.example.domain.customer.TaxPayments.LumpSumTax;
+import org.example.domain.invoice.IncomeInvoice;
+import org.example.domain.money.Currency;
+import org.example.domain.money.Money;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.YearMonth;
+
+import static org.example.application.CountHealthInsuranceContributionService.MINIMUM_WAGE;
+
+//@SpringJUnitConfig(classes = {ApplicationConfiguration.class})
+//@AllArgsConstructor(onConstructor = @__(@Autowired))
+class CountHealthInsuranceContributionServiceTest {
+//    @InjectMocks
+    private CountHealthInsuranceContributionService countHealthInsuranceContributionService;
+//    @Mock
+    private CountMonthlyMonthlyRevenueService countMonthlyRevenueService;
+//    @Mock
+    private CountMonthlyIncomeService countIncomeService;
+    private CountMonthlyCostsService countMonthlyCostsService;
+    private InsertIncomeInvoiceService insertIncomeInvoice;
+    private CountIncomeTaxService countIncomeTaxService;
+    private InMemoryInvoiceRepo invoiceRepo;
+    private InsertCostInvoiceService insertCostInvoice;
+    private InsertInvoiceService insertInvoiceService;
+
+    //problem z testami - Klasa CountIncomeService i countHealthInsuranceContributionService sa obie zalezne od siebie
```

Comment:
czyli cos zle zaimplementowane :)

Reply:
Here was just the comment to inform you what is the problem I would like to mention on our next meeting, but i resolved It on my own. -> Circular dependency issue.
Resolved